yt/yql/tests/test_simple.py

```diff
@@ -105,6 +105,58 @@ def test_calc(self, query_tracker, yql_agent):
         """, [[{"column0": 1}], [{"column0": 2}], [{"a": 1, "b": 2}]])
 
 
+class TestYqlAgentBan(TestQueriesYqlBase):
+    NUM_YQL_AGENTS = 1
+    YQL_AGENT_DYNAMIC_CONFIG = {"state_check_period": 2000}
+
+    def query_fails():
+        query = start_query("yql", 'select 1')
+        query.track()
+        return False
+
+    def query_completes():
+        query = start_query("yql", 'select 1')
+        try:
+            query.track()
+        except Exception:
+            return False
+        return True
+
+    @authors("mpereskokova")
+    def test_yql_agent_ban_on_new_queries(self, query_tracker, yql_agent):
+        address = yql_agent.yql_agent.addresses[0]
+        set(f"//sys/yql_agent/instances/{address}/@banned", True)
+
+        with raises_yt_error() as err:
+            wait(TestYqlAgentBan.query_fails)
+        assert err[0].contains_text("No alive peers found")
+
+        set(f"//sys/yql_agent/instances/{address}/@banned", False)
+
+        wait(TestYqlAgentBan.query_completes)
```

Comment:
is it intentional, that we don't check result value here (whether it failed or not)?

Reply:
In case of failure, it throws an error during testing - so if it completes successfully, I can be sure that there are no errors.

Reply:
now I got it!